### PR TITLE
Add roles and permission resources 

### PR DESCRIPTION
**Pull request by sumitAgrawal007**
### Description

Added Resource and DataSource to create and manage vsphere Roles.
Added Resource to create and manage vsphere permissions.

### Acceptance tests
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch? (If so, please include the test log in a gist)
Acceptance test logs: https://gist.github.com/sumitAgrawal007/c536fd77ae856af4424cd0bf3a86fb9d

### References

https://github.com/hashicorp/terraform-provider-vsphere/issues/1102

### CODE REVIEW

File: website/docs/r/vsphere_entity_permissions.html.markdown [hunk 1]
```diff
@@ -0,0 +1,78 @@
+---
+subcategory: "Administration Resources"
+layout: "vsphere"
+page_title: "VMware vSphere: Entity Permissions"
+sidebar_current: "docs-vsphere-entity-permissions"
+description: |-
+  Provides CRUD operations on a vsphere entity permissions. Permissions can be created on an entity for a given user
```
Comment (shylasrinivas) [inline website/docs/r/vsphere_entity_permissions.html.markdown:7]: user or group

File: vsphere/data_source_vsphere_role_test.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+resource "vsphere_role" test-role {
+  name = "terraform-test-role1"
+  role_privileges = ["%s", "%s","%s","%s"]
+}
+
+data "vsphere_role" "role1" {
+  label = vsphere_role.test-role.label
+}
+`,
+		PRIVILEGE_1,
```
Comment (shylasrinivas) [inline vsphere/data_source_vsphere_role_test.go:61]: where are these constants defined?

Comment (sumitAgrawal007) [reply vsphere/data_source_vsphere_role_test.go:61]: in vsphere/resource_vsphere_role.go file.

**Comment by sumitAgrawal007** [comment]
@bill-rich Can you have a look at this.

**Comment by bill-rich** [comment]
@sumitAgrawal007 I will check it out today.

**Pull request merged**